seaworthy/definitions.py

```diff
@@ -52,16 +51,41 @@ def remove(self, **kwargs):
         self.helper.remove(self.inner(), **kwargs)
         self._inner = None
 
-    def __enter__(self):
+    def setup(self, helper=None):
+        """
+        Setup this resource so that is ready to be used in a test. If the
+        resource has already been created, this call does nothing.
+
+        For most resources, this just involes creating the resource in Docker.
+
+        :param helper:
+            The resource helper to use, if one was not provided when this
+            resource definition was created.
+        """
+        if self.created:
+            return
+
+        self.set_helper(helper)
         self.create()
```

Comment:
Perhaps this should `return self`? That would streamline the case where we want to set up and use the resource in one go. For example, `self.container = MyContainer().setup(helper)` in a test case or `return self.setup()` in `__enter__()`.